scripts/analyze-bundle/command.mjs

```diff
@@ -0,0 +1,151 @@
+  }
+
+  await traverseDirectory(directory);
+  return files;
+}
+
+async function parseBuildJsonFiles(files) {
+  const packagesAndBundleSize = {};
+
+  for (const build of files) {
+    const data = JSON.parse(await fs.readFile(build, 'utf8'));
+    const outputBytes = Object.values(data.outputs)
+      .map((outputs) => outputs.bytes)
+      .reduce((prev, item) => prev + item, 0);
+    packagesAndBundleSize[path.basename(build)] = outputBytes;
+  }
+
+  return packagesAndBundleSize;
+}
+
+async function compareBuildJsonFiles(targetBranchBuilds, currentBranchBuilds) {
+  const targetBranchPackagesAndBundleSize = await parseBuildJsonFiles(
+    targetBranchBuilds
+  );
+  const currentBranchPackagesAndBundlesSize = await parseBuildJsonFiles(
+    currentBranchBuilds
+  );
+
+  const deletedPackage = Object.keys(targetBranchPackagesAndBundleSize).filter(
+    (item) => !Object.keys(currentBranchPackagesAndBundlesSize).includes(item)
+  );
+
+  const addedPackages = Object.keys(currentBranchPackagesAndBundlesSize).filter(
+    (item) => !Object.keys(targetBranchPackagesAndBundleSize).includes(item)
+  );
+
+  const changes = {};
+
+  for (const key in targetBranchPackagesAndBundleSize) {
+    const bundleSizeDiff =
+      currentBranchPackagesAndBundlesSize[key] -
+      targetBranchPackagesAndBundleSize[key];
+
+    changes[getPackageName(key)] = {
+      'current branch': formatBundleSize(
+        currentBranchPackagesAndBundlesSize[key]
+      ),
+      'target branch': formatBundleSize(targetBranchPackagesAndBundleSize[key]),
+      diff: formatBundleSizeDiff(
+        bundleSizeDiff,
+        targetBranchPackagesAndBundleSize[key]
+      ),
+    };
+  }
+
+  addedPackages.forEach((item) => {
+    changes[getPackageName(item)] = {
+      'current branch': formatBundleSize(
+        currentBranchPackagesAndBundlesSize[item]
+      ),
+      'target branch': formatBundleSize(0),
+      diff: formatBundleSizeDiff(currentBranchPackagesAndBundlesSize[item], 0),
+    };
+  });
+
+  deletedPackage.forEach((item) => {
+    changes[getPackageName(item)] = {
+      'current branch': formatBundleSize(0),
+      'target branch': formatBundleSize(
+        targetBranchPackagesAndBundleSize[item]
+      ),
+      diff: formatBundleSizeDiff(
+        -targetBranchPackagesAndBundleSize[item],
+        targetBranchPackagesAndBundleSize[item]
+      ),
+    };
+  });
+
+  return changes;
+}
+
+function getPackageName(fileName) {
+  return `@rango-dev/${fileName.split('.build.json')[0]}`;
+}
+
+async function run() {
+  const currentDirectory = process.cwd();
+  const targetBranchBuilds = await findBuildJsonFiles(currentDirectory, [
+    'current-branch',
+  ]);
+  const currentBranchBuilds = await findBuildJsonFiles(
+    path.join(currentDirectory, 'current-branch')
+  );
+
+  const comparison = await compareBuildJsonFiles(
+    targetBranchBuilds,
+    currentBranchBuilds
+  );
+  console.table(comparison);
+}
+
+run().catch((error) =>
+  console.error(
```

Comment:
only using `console.error` isn't enough. because it means process successfully completed for node. you need to add `process.exit(1);` to signal something went wrong. it's important because actions are rely on this process status codes.

Reply:
You can see other commands and copy pasta